src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1495,7 +1495,11 @@ private IEnumerable<PSTypeName> InferTypesFrom(MemberExpressionAst memberExpress
             }
 
             var res = new List<PSTypeName>(10);
-            bool isInvokeMemberExpressionAst = memberExpressionAst is InvokeMemberExpressionAst;
+
+            var invokeMemberExpression = memberExpressionAst as InvokeMemberExpressionAst;
+            bool isInvokeMemberExpressionAst = invokeMemberExpression is not null;
```

Comment:
```c#
bool isInvokeMemberExpressionAst = memberExpressionAst is InvokeMemberExpressionAst;
```
Doesn't this work here?

Reply:
`invokeMemberExpression` is new and I don't see it is used in other places.

Reply:
Refactored this code a little bit to address @iSazonov's comment.